src/services/createPropStore.ts

```diff
@@ -8,37 +8,109 @@ export const propStoreKey: InjectionKey<PropStore> = Symbol()
 
 type ComponentProps = { id: string, name: string, props?: (params: Record<string, unknown>) => unknown }
 
+type RoutePropsContext = {
+  /**
+   * The unique route id generated by `createRouteId`
+   */
+  id: string,
+  /**
+   * The named view the component was assigned to
+   */
+  name: string,
+  /**
+   * The specific params for the actual route
+   */
+  params: unknown,
+}
+
+/**
+ * A unique identifier for a component's props based on the route id, view name, and stringified params
+ * Format: `${id}-${name}-${params}
+ */
+type RoutePropsKey = `${string}-${string}-${string}`
```

Comment:
I considered this as well, but felt like it was over engineering since we never later infer out any of the individual pieces